libr/fs/fs.c

```diff
@@ -700,13 +702,15 @@ R_API int r_fs_prompt(RFS* fs, const char* root) {
 	}
 
 	for (;;) {
-		printf ("[%s]> ", path);
-		fflush (stdout);
-		fgets (buf, sizeof (buf) - 1, stdin);
-		if (feof (stdin)) {
-			break;
-		}
-		buf[strlen (buf) - 1] = '\0';
+		strncpy (prompt, path, PROMT_PATH_BUFSIZE);
+		strcat (prompt, "> ");
+		shell->set_prompt (prompt);
+		ptr = shell->readline ();
+		shell->hist_add (ptr);
+
+		if (!ptr) break;
+
+		strncpy (buf, ptr, PROMT_PATH_BUFSIZE);
```

Comment:
In this case, I think it's useful to keep that copy, because later in other places there are copies made considering sizeof(buf) that wouldn't work with sizeof(ptr), as ptr is const char*. But will fix it too.